### PR TITLE
Adds being British as a negative quirk.

### PR DESCRIPTION
**Pull request by AnCopper**
## About The Pull Request

You can now select to be British as a negative quirk.

By the way, someone please come up with funnier messages for gaining/losing this quirk, I couldn't think of anything.

## Why It's Good For The Game

I got this in one round as a mutation after getting cloned, and this is honestly the funniest shit that I have seen in my 4 years of playing this game. The reason why I added it as a negative quirk is because it makes your speech partially unintelligible at times.

So, saying: 
"Captain, I need a new ID from the Head of Personnel, a traitor stole it, and security isn't doing anything about it"
gets turned into
"Queen, MAN need a new BBM PIN from 'he Head of Badmannel, a roadman s'ole i', and like copperz isn'' doin' any'hin' 'bou' i'"

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl: EvilDragonfiend Ancopper
add: Added being British as a negative quirk.
/:cl:

### CODE REVIEW

**Comment by ToasterBan** [comment]
this is what bee needed

**Comment by EvilDragonfiend** [comment]
It looks it's removable and just a free negative point. You need this mutation unremovable.
or at least you need to make it a neutral quirk if you don't want to make it unremovable.

**Comment by AnCopper** [comment]
I can't think of any way to make a mutation permanent.

**Comment by EvilDragonfiend** [comment]
@Ivniinvi I think you know how the mutation code work.

**Comment by EvilDragonfiend** [comment]
I took a look at the mutation code and it is the CHAV mutation code
```
/datum/mutation/chav/on_acquiring(mob/living/carbon/owner)
	if(..())
		return
	RegisterSignal(owner, COMSIG_MOB_SAY, .proc/handle_speech)
```
So, apparently, you need to take it as - instead of giving a mutation in the quirk code:
```
/datum/quirk/british/add()
	var/mob/living/carbon/H = quirk_holder
	RegisterSignal(H, COMSIG_MOB_SAY, .proc/handle_speech)
```
Also, You need to put a removal presentation in the mutation code - something like this
```
/datum/mutation/chav/on_losing(mob/living/carbon/owner)
	if(..())
		return
	if( something condition that when filters the quirk hold "British")
		return
	UnregisterSignal(owner, COMSIG_MOB_SAY)
```

(I mentioned Ivniinvi to ask him for the mutation code, then removed mentioning him as I think I found a solution.)

**Comment by EvilDragonfiend** [comment]
ah, wait. there's a little mistake in my suggestion

**Comment by EvilDragonfiend** [comment]
it looks you need to copy `/datum/mutation/chav/proc/handle_speech(datum/source, list/speech_args)` to `/datum/quirk/british/proc/handle_speech(datum/source, list/speech_args)` to make my suggestion works.

**Comment by tasteofaliar** [comment]
Oh my god, Do you play David? I was the A.I on a round where a scientist on exploration comms had exploded into gibs and gotten cloned, and I could barely pay attention or understand to the british gibberish I kept glancing at while being forced to look at other things

10/10, great PR

**Review by ike709 — CHANGES_REQUESTED** [review]
Based. Needs more mechanical downsides if it's going to give you a point though.

**Comment by AnCopper** [comment]
> Based. Needs more mechanical downsides if it's going to give you a point though.

British people will now get sick from every food that normal people like, and only like tin beans.

**Comment by AnCopper** [comment]
Well, I have no idea why the compiler is failing here, but it works on my end.
image

**Comment by EvilDragonfiend** [comment]
> > Based. Needs more mechanical downsides if it's going to give you a point though.
> 
> British people will now get sick from every food except tin-canned beans.

I am curious how it would go with deviant tastes quirk...

**Comment by Howluinb** [comment]
add only beans nutrition, or pure bread. Only beans is really hard to survive off of because there's barely any on station, but bread is more manageable

https://en.wikipedia.org/wiki/Toast_sandwich

**Comment by AnCopper** [comment]
> add only beans nutrition, or pure bread. Only beans is really hard to survive off of because there's barely any on station, but bread is more manageable
> 
> https://en.wikipedia.org/wiki/Toast_sandwich

It only makes you puke if you eat stuff that your species would normally like.

And to be honest, I don't think that we need more punishment for a -1 point quirk, it already does more than some other ones.

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -671,3 +671,57 @@
...
+	species.disliked_food = initial(species.disliked_food)
+
+/datum/quirk/british/proc/handle_speech(datum/source, list/speech_args)
+	SIGNAL_HANDLER
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = " [message]"
+		var/list/whole_words = strings(BRIISH_TALK_FILE, "words")
+		var/list/british_sounds = strings(BRIISH_TALK_FILE, "sounds")
+		var/list/british_appends = strings(BRIISH_TALK_FILE, "appends")
```
Comment (ike709) [inline code/datums/traits/negative.dm:704]: Should probably be static lists.

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -671,3 +671,57 @@
...
+
+		for(var/key in whole_words)
```
Comment (ike709) [inline code/datums/traits/negative.dm:706]: These loops seem kind of expensive. There's not a better way to do this?

Comment (AnCopper) [reply code/datums/traits/negative.dm:706]: I just copied the existing code from the actual mutation.

File: code/datums/traits/negative.dm [hunk 3]
```diff
@@ -671,3 +671,57 @@
...
+	value = -1
+	gain_text = "<span class='danger'>Ye feel like a reet prat like, innit?.</span>"
+	lose_text = "<span class='notice'>You no longer feel like being rude and sassy.</span>"
+
+/datum/quirk/british/add()
+	var/mob/living/carbon/human/H = quirk_holder
+	RegisterSignal(H, COMSIG_MOB_SAY, .proc/handle_speech)
+	var/datum/species/species = H.dna.species
+	var/liked = species.liked_food
+	species.liked_food = /obj/item/reagent_containers/food/snacks/canned/beans
```
Comment (PowerfulBacon) [inline code/datums/traits/negative.dm]: Im pretty certain this is not how liked food works

**Comment by EvilDragonfiend** [comment]
funny fix

**Review by ike709 — APPROVED** [review]

**Comment by pixelman77** [comment]
where speedmerge

File: code/datums/traits/negative.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/datums/traits/negative.dm:704]: this review still applies

**Comment by PowerfulBacon** [comment]
I hate the copy and pasted code, is there not some way to make these both go through the same thing? If someone updates british code (which admitadly is kinda bad since it was copied from uwuspeak code) then it will require double the updates

**Comment by AnCopper** [comment]
> I hate the copy and pasted code, is there not some way to make these both go through the same thing? If someone updates british code (which admitadly is kinda bad since it was copied from uwuspeak code) then it will require double the updates

I don't think so, otherwise it would simply be a removable mutation.

**Comment by BrumalLake** [comment]
free mime point

**Comment by AnCopper** [comment]
> free mime point

The beans modifier still affects them.

**Comment by AnCopper** [comment]
Cranky_RockingChair

File: code/datums/traits/negative.dm [hunk 4]
```diff
@@ -671,3 +671,57 @@
...
+			if(islist(value))
+				value = pick(value)
+
+			message = replacetextEx(message, "[uppertext(key)]", "[uppertext(value)]")
+			message = replacetextEx(message, "[capitalize(key)]", "[capitalize(value)]")
+			message = replacetextEx(message, "[key]", "[value]")
+
+		if(prob(8))
+			message += pick(british_appends)
+	speech_args[SPEECH_MESSAGE] = trim(message)
```
Comment (PowerfulBacon) [inline code/datums/traits/negative.dm:726]: Clean up this code, use static lists when reading from huge text files.
Also make this a proc and don't have duplicated code

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

**Comment by PowerfulBacon** [comment]
Review comments not addressed and while this is funny, there is nothing else really going for it.

**Pull request closed without merging**